### PR TITLE
Fixed #67

### LINKED ISSUE #67: Uncaught exception "Illegal unescaped char"
We have recently migrated from node 8 to node 12 and due to the socket problems of ldapjs we switched from activedirectory module to your activedirectory2@2.0.0-pre.1 because it is maintained and has the necessary fixes to run with node >= 10.

Unfortunately we have one user who crashes the process when he loggs in and his AD groups are retrieved:

`
Uncaught exception: {"name":"Error","message":"illegal unescaped char: (","stack":"Error: illegal unescaped char: (\n    at escapeValue (/home/app/node_modules/ldap-filter/lib/index.js:49:13)\n    at parseExpr (/home/app/node_modules/ldap-filter/lib/index.js:167:16)\n    at parseFilter (/home/app/node_modules/ldap-filter/lib/index.js:228:14)\n    at Object.parse (/home/app/node_modules/ldap-filter/lib/index.js:252:18)\n    at Object.parseString (/home/app/node_modules/ldapjs/lib/filters/index.js:180:27)\n    at Client.search (/home/app/node_modules/ldapjs/lib/client/client.js:566:30)\n    at Searcher.search (/home/app/node_modules/activedirectory2/lib/components/search.js:118:15)\n    at search (/home/app/node_modules/activedirectory2/lib/components/search.js:352:19)\n    at GroupMembersForDN.getMembers (/home/app/node_modules/activedirectory2/lib/components/getGroupMembershipForDN.js:111:3)\n    at getGroupMembershipForDN (/home/app/node_modules/activedirectory2/lib/components/getGroupMembershipForDN.js:157:17)\n    at ActiveDirectory.ggmfd [as getGroupMembershipForDN] (/home/app/node_modules/activedirectory2/lib/activedirectory.js:292:10)\n    at /home/app/node_modules/activedirectory2/lib/activedirectory.js:489:10\n    at /home/app/node_modules/activedirectory2/lib/activedirectory.js:380:5\n    at Searcher.callback (/home/app/node_modules/activedirectory2/lib/activedirectory.js:340:5)\n    at Searcher.onSearchEnd (/home/app/node_modules/activedirectory2/lib/components/search.js:283:12)\n    at EventEmitter.<anonymous> (/home/app/node_modules/activedirectory2/lib/components/search.js:149:12)"}`

At the moment I do not have any more insights and I guess it will be hard finding out which is possibly the group name which leads to the crash.

Do you have any idea?

Until we can fix this issue we're unfortunately forced to go back to the "old" version using activedirectoy with a scheduled hourly restart of the process in order not to exceed the max. number of file descriptors (sockets) per process on our linux server.

### PR DESCRIPTION
**Pull request by Frank683**
Hey @jsumners 

Please review the fix.
Please also note that I removed the **"** from

`char = '\\\\2A"'`

on line 287 because I think it has slipped in there unintentionally.

### CODE REVIEW

File: test/utilities.test.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+const utilities = require('../lib/components/utilities')
+
+tap.test('Utility functions', t => {
+  t.beforeEach((done, t) => {
+    done()
+  })
+
+  t.afterEach((done, t) => {
+    done()
+  })
```
Comment (jsumners) [inline test/utilities.test.js:13]: These are unnecessary.

Comment (Frank683) [reply test/utilities.test.js:13]: Yes, in the current state indeed. I left them in as template for someone who might add tests to this module in the future. But I can remove them for now of course.

Comment (jsumners) [reply test/utilities.test.js:13]: Please. I'd rather not add code for _potential_ future cases. The code should reflect the current state.

Comment (Frank683) [reply test/utilities.test.js]: The unnecessary code has been removed.

**Comment by jsumners** [comment]
Published under `activedirectory2@next`.

**Pull request merged**